source/server/lds_api.cc

```diff
@@ -97,7 +97,7 @@ void LdsApiImpl::onConfigUpdate(const std::vector<Config::DecodedResourceRef>& r
   // We need to keep track of which listeners need to remove.
   // Specifically, it's [listeners we currently have] - [listeners found in the response].
   absl::node_hash_set<std::string> listeners_to_remove;
-  for (const auto& listener : listener_manager_.listeners()) {
```

Comment:
This is interesting. There are 3 cases.
1. The name is in active, but not in warm;
2. The name is not in active, but in warm;
3. The name is in both active and warm.

Here we put the name in the to_remove list, what we also need to check is that the ListenerManager is able to remove the listener if it's in warm but not in active (2nd case)

Reply:
https://github.com/envoyproxy/envoy/blob/6eb7e642d33f5a55b63c367188f09819925fca34/source/server/listener_manager_impl.cc#L828-L836

Confirmed ListenerManagerImpl::removeListener is doing well ^^

Reply:
In what case wouldn't the ListenerManager be able to remove a listener that's in warm but not active?